src/views/SalesAdd.vue

```diff
@@ -0,0 +1,200 @@
+<template>
+  <v-container>
+    <v-sheet width="300" class="mx-auto">
+      <div class="mx-auto text-center" style="font-size: 24px">
+        {{ pageTitle }}
+      </div>
+      <v-form @submit.prevent="handleSubmit" ref="form">
+        <v-select
+          label="Seller"
+          v-model="seller"
+          :items="allUsersNames"
+          :rules="[validateAuthors(seller)]"
+        >
+        </v-select>
+
+        <v-select
+          v-model="buyer"
+          label="Buyer"
+          :items="allUsersNames"
+          :rules="[validateAuthors(buyer)]"
+        >
+        </v-select>
+
+        <v-container class="d-flex justify-space-between">
+          <v-btn
+            color="primary"
+            @click="toggleDialog(true)"
+            :disabled="!selectedUser"
+          >
+            Products
+          </v-btn>
+
+          <v-btn
+            color="green lighten-1"
+            @click="() => (calendarDialog = true)"
+            :disabled="!selectedUser"
+            >Calendar</v-btn
+          >
+        </v-container>
+
+        <v-container v-if="productsContent.length > 0">
+          <div class="mx-auto text-center" style="font-size: 16px">
+            Currently selected products:
+          </div>
+          <ProductRow
+            v-for="(product, i) in productsContent"
+            :key="product"
+            :product="productsContent[i]"
+          />
+        </v-container>
+
+        <v-container class="d-flex flex-column mt-4">
+          <p>Total amount: € {{ totalAmount.toFixed(2) || 0 }}</p>
+          <p>Discounted amount: € {{ discountedAmount.toFixed(2) || 0 }}</p>
+          <p>Total discount: {{ totalDiscount.toFixed(2) || 0 }} %</p>
+        </v-container>
+
+        <submit-buttons @reset-form="resetForm" />
+      </v-form>
+    </v-sheet>
+    <products-dialog
+      v-if="selectedUser"
+      v-model="productsDialog"
+      @close-dialog="toggleDialog(false)"
+      @save-product-dialog="productsTableValues"
+      :userId="selectedUser.id"
+    >
+    </products-dialog>
+
+    <calendar-dialog
+      v-model="calendarDialog"
+      @close-dialog="handleCloseCalendar"
+    />
+  </v-container>
+</template>
+
+<script setup>
+import CalendarDialog from "../components/Dialog/CalendarDialog.vue";
+import ProductRow from "../components/ProductRow.vue";
+import { validateAuthors } from "../utils/validation-rules";
+import ProductsDialog from "@/components/Dialog/ProductsDialog.vue";
+import { useRoute, useRouter } from "vue-router";
+import { useStore } from "vuex";
+import { ref, computed, watch, inject } from "vue";
+const snackbarProvider = inject("snackbarProvider");
+const [route, router] = [useRoute(), useRouter()];
+const store = useStore();
+const pageTitle = ref(route.meta.title);
+const [seller, buyer] = [ref([]), ref([])];
+const form = ref(null);
+const selectedUser = ref("");
+const [productsDialog, productsContent] = [ref(false), ref([])];
```

Comment:
`productsContent` could be renamed to `productsForSale` to be more precise

Reply:
fixed